File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
  let user = {};
  var answer = [0];
  const ban_list = [];
  report = [...new Set(report)];

  console.log(report);
  id_list.forEach((id, idx) => {
    user[id] = 0;
    answer[idx] = 0;
  });
  for (rep of report) {
    rep = rep.split(" ");
    user[rep[1]]++;
  }

  for (user_id of Object.keys(user)) {
    if (user[user_id] >= k) {
      ban_list.push(user_id);
    }
  }

  for (rep of report) {
    rep = rep.split(" ");
    if (ban_list.includes(rep[1])) {
      answer[id_list.indexOf(rep[0])] += 1;
    }
  }

  console.log(user);
  console.log(ban_list);
  return answer;
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.Map;
import java.util.HashMap;
import java.util.Set;
import java.util.HashSet;
import java.util.StringTokenizer;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        HashMap<String, Set<String>> reportMap = new HashMap<>(); 
        HashMap<String, Integer> mailMap = new HashMap<>();
        
        for(int i = 0; i < report.length; i++){
            StringTokenizer st = new StringTokenizer(report[i]);
            String reporter = st.nextToken();
            String reported = st.nextToken();
            
            if(!reportMap.containsKey(reported)){
                reportMap.put(reported, new HashSet<String>());
                reportMap.get(reported).add(reporter);
            }else {
                reportMap.get(reported).add(reporter);
            }
            
        
        }
        
        for(int i = 0; i < id_list.length; i++){
            if(reportMap.containsKey(id_list[i])){
                if(reportMap.get(id_list[i]).size() >= k){
                    
                    for(int j = 0; j < id_list.length; j++){
                        if(reportMap.get(id_list[i]).contains(id_list[j])){
                            if(!mailMap.containsKey(id_list[j])){
                                mailMap.put(id_list[j], 1);
                            }else{
                                mailMap.put(id_list[j], mailMap.get(id_list[j])+1);
                            }
                        }
                    }
                }
            }  
        }
        
        
        
        for(int i = 0; i < id_list.length; i++){
            if(!mailMap.containsKey(id_list[i]))
                answer[i] = 0;
            else
                answer[i] = mailMap.get(id_list[i]);
        }
        
        return answer;
    }
}